File: server/php/src/Infrastructure/Mappers/HostedTokenizationMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers;

use OnlinePayments\ExampleApp\Application\DTOs\HostedTokenization\ResponseDto;
use OnlinePayments\Sdk\Domain\CreateHostedTokenizationResponse;

class HostedTokenizationMapper
{
    public static function mapFromSdkResponse(?CreateHostedTokenizationResponse $response): ResponseDto
    {
        return new ResponseDto(
            $response?->getHostedTokenizationId() ?? null,
            $response?->getHostedTokenizationUrl() ?? null
        );
    }
}

File: server/python/app/infrastructure/mappers/hosted_tokenization_mapper.py
from onlinepayments.sdk.domain.create_hosted_tokenization_response import CreateHostedTokenizationResponse
from app.application.dtos.hosted_tokenization.response_dto import HostedTokenizationResponseDto

class HostedTokenizationMapper:

    @staticmethod
    def from_sdk_response(response: CreateHostedTokenizationResponse) -> HostedTokenizationResponseDto:
        return HostedTokenizationResponseDto(
            hosted_tokenization_id=response.hosted_tokenization_id if response else None,
            hosted_tokenization_url=response.hosted_tokenization_url if response else None
        )